src/functions/artifacts.spec.ts

```diff
@@ -0,0 +1,318 @@
+      const repo: artifactregistry.Repository = {
+        name: "projects/my-project/locations/us-central1/repositories/gcf-artifacts",
+        format: "DOCKER",
+        description: "Cloud Functions container artifacts",
+        createTime: "",
+        updateTime: "",
+        cleanupPolicies: {
+          "other-policy": {
+            id: "other-policy",
+            action: "DELETE",
+            condition: {
+              tagState: "ANY",
+              olderThan: "432000s",
+            },
+          },
+        },
+      };
+
+      const policy = artifacts.findExistingPolicy(repo);
+      expect(policy).to.be.undefined;
+    });
+
+    it("should return the policy if it exists", () => {
+      const expectedPolicy: artifactregistry.CleanupPolicy = {
+        id: artifacts.CLEANUP_POLICY_ID,
+        action: "DELETE",
+        condition: {
+          tagState: "ANY",
+          olderThan: "432000s",
+        },
+      };
+
+      const repo: artifactregistry.Repository = {
+        name: "projects/my-project/locations/us-central1/repositories/gcf-artifacts",
+        format: "DOCKER",
+        description: "Cloud Functions container artifacts",
+        createTime: "",
+        updateTime: "",
+        cleanupPolicies: {
+          [artifacts.CLEANUP_POLICY_ID]: expectedPolicy,
+        },
+      };
+
+      const policy = artifacts.findExistingPolicy(repo);
+      expect(policy).to.deep.equal(expectedPolicy);
+    });
+  });
+
+  describe("daysToSeconds", () => {
+    it("should convert days to seconds with 's' suffix", () => {
+      expect(artifacts.daysToSeconds(1)).to.equal("86400s");
+      expect(artifacts.daysToSeconds(5)).to.equal("432000s");
+      expect(artifacts.daysToSeconds(30)).to.equal("2592000s");
+    });
+  });
+
+  describe("parseDaysFromPolicy", () => {
+    it("should correctly parse seconds into days", () => {
+      expect(artifacts.parseDaysFromPolicy("86400s")).to.equal(1);
+      expect(artifacts.parseDaysFromPolicy("432000s")).to.equal(5);
+      expect(artifacts.parseDaysFromPolicy("2592000s")).to.equal(30);
+    });
+
+    it("should return undefined for invalid formats", () => {
+      expect(artifacts.parseDaysFromPolicy("5d")).to.be.undefined;
+      expect(artifacts.parseDaysFromPolicy("invalid")).to.be.undefined;
+      expect(artifacts.parseDaysFromPolicy("")).to.be.undefined;
+    });
+  });
+
+  describe("generateCleanupPolicy", () => {
+    it("should generate a valid cleanup policy with the correct days", () => {
+      const policy = artifacts.generateCleanupPolicy(7);
+      expect(policy).to.have.property(artifacts.CLEANUP_POLICY_ID);
+      expect(policy[artifacts.CLEANUP_POLICY_ID].id).to.equal(artifacts.CLEANUP_POLICY_ID);
+      expect(policy[artifacts.CLEANUP_POLICY_ID].action).to.equal("DELETE");
+      expect(policy[artifacts.CLEANUP_POLICY_ID].condition).to.deep.include({
+        tagState: "ANY",
+        olderThan: "604800s", // 7 days in seconds
```

Comment:
nit: I often find it easier to write something like `${60 * 24 * 7}s` to make it clear that the math is right.